### PR TITLE
Add tips for troubleshooting registry auth

### PR DESCRIPTION
**Pull request by chmeliik**

### CODE REVIEW

**Review by MartinBasti — APPROVED** [review]

File: docs/modules/ROOT/pages/troubleshooting/index.adoc [hunk 1]
```diff
@@ -44,4 +44,110 @@ If you see this error message, it means that the pipeline run has exceeded the d
...
\ No newline at end of file
+----
+
+== Failed to push or pull image
+
+Failing to authenticate with a container registry can be hard to debug. To be able to troubleshoot
+effectively, one needs to understand how registry authentication works in Tekton. For the full
+details, see the Tekton link:https://tekton.dev/docs/pipelines/auth/[Authentication] docs. In short:
+
+. Tekton uses a ServiceAccount (named `appstudio-pipeline` by default in Konflux) to run your Pipeline.
```
Comment (arewm) [inline docs/modules/ROOT/pages/troubleshooting/index.adoc:55]: Does Tekton use this SA or does {ProductName}?

File: docs/modules/ROOT/pages/troubleshooting/index.adoc [hunk 2]
```diff
@@ -44,4 +44,110 @@ If you see this error message, it means that the pipeline run has exceeded the d
 spec:
   # ...
   timeout: 1h30m0s  # increase accordingly
-----
\ No newline at end of file
+----
+
+== Failed to push or pull image
```
Comment (arewm) [inline docs/modules/ROOT/pages/troubleshooting/index.adoc:49]: Replace Konflux with `{ProductName}`.

File: docs/modules/ROOT/pages/troubleshooting/index.adoc [hunk 3]
```diff
@@ -44,4 +44,110 @@ If you see this error message, it means that the pipeline run has exceeded the d
...
+
+Failing to authenticate with a container registry can be hard to debug. To be able to troubleshoot
+effectively, one needs to understand how registry authentication works in Tekton. For the full
+details, see the Tekton link:https://tekton.dev/docs/pipelines/auth/[Authentication] docs. In short:
+
+. Tekton uses a ServiceAccount (named `appstudio-pipeline` by default in Konflux) to run your Pipeline.
+. This ServiceAccount has a list of `imagePullSecrets` and `secrets`.
+.. Tekton uses the `imagePullSecrets` when pulling the images that execute the Tasks in your pipeline.
+   These images are typically hardcoded in the Task definition and publicly accessible. As a Konflux
+   user, you're more interested in the `secrets`.
```
Comment (arewm) [inline docs/modules/ROOT/pages/troubleshooting/index.adoc:59]: Are they more interested in this case or generally?

File: docs/modules/ROOT/pages/troubleshooting/index.adoc [hunk 4]
```diff
@@ -44,4 +44,110 @@ If you see this error message, it means that the pipeline run has exceeded the d
...
+----
+secret_name=secret-for-my-registry
+
+oc secret link appstudio-pipeline "$secret_name"
+----
+
+=== Get the merged registry auth file
+
+This script roughly approximates Tekton's merging of `dockercfg` and `dockerconfigjson` secrets.
+It does not handle `basic-auth` secrets.
```
Comment (arewm) [inline docs/modules/ROOT/pages/troubleshooting/index.adoc:128]: Does this script have any prerequisites? Platform requirements?

File: docs/modules/ROOT/pages/troubleshooting/index.adoc (hunk 3 above)
Comment (chmeliik) [reply docs/modules/ROOT/pages/troubleshooting/index.adoc:59]: Almost always. The only time `imagePullSecrets` are relevant is if they're developing their own Task and using a private image as the `image:` to run a step

Or if Konflux devs screw up and use a private image for a Konflux task

File: docs/modules/ROOT/pages/troubleshooting/index.adoc (hunk 1 above)
Comment (chmeliik) [reply docs/modules/ROOT/pages/troubleshooting/index.adoc:55]: Both I guess? Tekton uses the SA, Konflux calls it `appstudio-pipeline`

File: docs/modules/ROOT/pages/troubleshooting/index.adoc (hunk 4 above)
Comment (chmeliik) [reply docs/modules/ROOT/pages/troubleshooting/index.adoc:128]: `jq`, `kubectl` and read access to Secrets (same as the other script)

Comment (arewm) [reply docs/modules/ROOT/pages/troubleshooting/index.adoc:128]: We have requirements specified for procedures in various files. Can we just add a quick call-out? The CLI and Admin access are likely more interesting pre-requisites.

Comment (chmeliik) [reply docs/modules/ROOT/pages/troubleshooting/index.adoc:132]: Added

**Review by arewm — APPROVED** [review]

**Pull request merged**